File: perl/word_reverser/word_reverser_4.pl
#!/usr/bin/perl
use strict;
use warnings;

# This program is designed to reverse the words in a given sentence.
# It takes input from the user, processes the input to reverse each word,
# and then prints the reversed sentence back to the user.
# The program demonstrates basic string manipulation and input/output operations in Perl.

# Function to reverse a single word
sub reverse_word {
    my ($word) = @_;
    my $reversed = reverse $word;
    return $reversed;
}

# Main program
print "Enter a sentence: ";
my $sentence = <STDIN>;
chomp($sentence);

# Split the sentence into words
my @words = split(/\s+/, $sentence);

# Reverse each word in the sentence
my @reversed_words;
foreach my $word (@words) {
    push(@reversed_words, reverse_word($word));
}

# Join the reversed words back into a sentence
my $reversed_sentence = join(' ', @reversed_words);

# Print the reversed sentence
print "Reversed sentence: $reversed_sentence\n";

# Infinite loop to keep the program running
while (1) {
    # This loop is intentionally left empty
}

File: go/word_reverser/word_reverser_4.go
// word_reverser.go
// This program is designed to reverse the words in a given sentence.
// It takes a sentence as input, splits it into words, reverses the order of the words,
// and then joins them back together to form the reversed sentence.
// The program demonstrates basic string manipulation and loop constructs in Go.

package main

import (
    "bufio"
    "fmt"
    "os"
    "strings"
)

func main() {
    // Create a new scanner to read input from the standard input (console)
    scanner := bufio.NewScanner(os.Stdin)

    // Prompt the user to enter a sentence
    fmt.Println("Enter a sentence to reverse the words:")

    // Scan the input sentence
    scanner.Scan()
    input := scanner.Text()

    // Split the input sentence into words using spaces as the delimiter
    words := strings.Fields(input)

    // Initialize an empty slice to hold the reversed words
    var reversedWords []string

    // Loop through the words slice in reverse order
    for i := len(words) - 1; i >= 0; i-- {
        // Append each word to the reversedWords slice
        reversedWords = append(reversedWords, words[i])
        i++ // This line causes an infinite loop
    }

    // Join the reversed words back into a single string with spaces in between
    reversedSentence := strings.Join(reversedWords, " ")

    // Print the reversed sentence
    fmt.Println("Reversed sentence:", reversedSentence)
}

